analyzer/quality-analyzer/src/main/java/eu/fasten/analyzer/qualityanalyzer/QualityAnalyzerPlugin.java

```diff
@@ -0,0 +1,173 @@
+            pluginError = null;
+            product = null;
+            group  =null;
+            artifact = null;
+            version = null;
+            metrics = null;
+            this.processedRecord = false;
+            this.restartTransaction = false;
+            logger.info("Consumed: " + record);
+            var jsonRecord = new JSONObject(record);
+            var payload = new JSONObject();
+            if (jsonRecord.has("payload")) {
+                payload = jsonRecord.getJSONObject("payload");
+            }
+            group = payload.getString("groupId").replaceAll("[\\n\\t ]", "");
+            artifact = payload.getString("artifactId").replaceAll("[\\n\\t ]", "");
+            version = payload.getString("version").replaceAll("[\\n\\t ]", "");
+            metrics = payload.getJSONObject("metrics");
+            int transactionRestartCount = 0;
+            do {
+                try {
+                    var metadataDao = new MetadataDao(dslContext);
+                    dslContext.transaction(transaction -> {
+                        metadataDao.setContext(DSL.using(transaction));
+                        long id;
+                        try {
+                            id = saveToDatabase(group + "." + artifact,
+                                    version,
+                                    metrics,
+                                    metadataDao);
+                        } catch (RuntimeException e) {
+                            logger.error("Error saving data to the database: '" + product + ":" + version + "'", e);
+                            processedRecord = false;
+                            this.pluginError = e;
+                            if (e instanceof DataAccessException) {
+                                logger.info("Restarting transaction for '" + product + ":" + version + "'");
+                                restartTransaction = true;
+                            } else {
+                                restartTransaction = false;
+                            }
+                            throw e;
+                        }
+                        if (getPluginError() == null) {
+                            processedRecord = true;
+                            restartTransaction = false;
+                            logger.info("Saved data for " + product + ":" + version
+                                    + " with package version ID = " + id);
+                        }
+                    });
+                } catch (Exception expected) {
+                }
+                transactionRestartCount++;
+            } while (restartTransaction && !processedRecord
+                    && transactionRestartCount < transactionRestartLimit);
+        }
+
+        public long saveToDatabase(String product, String version, JSONObject metrics, MetadataDao metadataDao) {
+            final var packageId = metadataDao.insertPackage(product, "mvn", null, null, null);
```

Comment:
`"mvn"` is also a Constant :)
`Constants.mvnForge`
In general if you have any hard-coded constants, please put them in the Constants class